src/controller.rs

```diff
@@ -349,10 +349,36 @@ pub async fn run_cluster_controller(state: State) {
         .default_backoff()
         .for_each(|_| futures::future::ready(()));
 
+    let capi_clusters = metadata_watcher(
+        Api::<Cluster>::all(client.clone()), 
+        Config::default().any_semantic(),
+    )
+    .default_handling();
+
+    let (sub, reader) = state.dispatcher.subscribe();
+    let cluster_ns_controller = Controller::for_shared_stream(sub, reader.clone())
+        .watches_stream(capi_clusters, move |_cluster| {
+            reader
+                .state()
+                .into_iter()
+                .filter_map(move |c: Arc<Cluster>| {
+                    info!("Reconcile CAPI cluster: {}", c.name_any());
+                    Some(ObjectRef::from_obj(&*c))
+                })
+        })
```

Comment:
```suggestion

    let (sub, reader) = state.dispatcher.subscribe();
    let cluster_ns_controller = Controller::for_shared_stream(sub, reader)
```
This does not skip reconcile on added cluster @yiannistri. The reason is simple - `multi_dispatcher` already streams clusters which match label selector or clusters in namespaces which match label selector.

However as I mentioned - this controller can miss removal event if it is down. Therefore integrating this logic to the `to_bundle` and `sync` path would handle it for you.

We also send patch requests on each reconcile for the namespace resource in the `reconcile_fleet_annotation_in_capi_ns`. This is not needed and may add empty load on the API server.